modules/authentication.py

```diff
@@ -185,24 +186,24 @@ def verify_response(
         # Macaroon not valid anymore, needs refresh
         if verified['account'] is None:
             empty_session(session)
-            return {
-                'status_code': 307,
-                'redirect': url_login,
-                'reason': 'Need login'
-            }
+            raise ApiVerifyResponseError(
+                'Need login',
+                307,
+                url_login
+            )
 
         # Not authorized content
         if response.status_code == 401 and not verified['allowed']:
-            return {
-                'status_code': 401,
-                'redirect': url_account,
-                'reason': 'Not authorized'
-            }
+            raise ApiVerifyResponseError(
+                'Not authorized',
+                401,
```

Comment:
Could you write this as `status_code=response.status_code`? Firstly, I think it's helpful to use named arguments if you can because it helps to clarify how the function might be using that argument. Secondly, I think we should make it clear that this API library is never simply choosing status_codes to pass back itself - whenever an ApiError has a status_code property it's because that status_code came directly from the API itself.